Sources/SemanticIndex/SemanticIndexManager.swift

```diff
@@ -279,20 +290,20 @@ public final actor SemanticIndexManager {
   /// The returned task finishes when all files are indexed.
   @discardableResult
   private func index(files: some Collection<DocumentURI>, priority: TaskPriority?) async -> Task<Void, Never> {
-    let outOfDateFiles = files.filter {
-      if case .upToDate = indexStatus[$0] {
+    let outOfDateFiles = await filesToIndex(toCover: files).filter {
+      if case .upToDate = indexStatus[$0.uri] {
         return false
       }
       return true
     }
-    .sorted(by: { $0.stringValue < $1.stringValue })  // sort files to get deterministic indexing order
+    .sorted(by: { $0.uri.stringValue < $1.uri.stringValue })  // sort files to get deterministic indexing order
 
     // Sort the targets in topological order so that low-level targets get built before high-level targets, allowing us
     // to index the low-level targets ASAP.
-    var filesByTarget: [ConfiguredTarget: [DocumentURI]] = [:]
+    var filesByTarget: [ConfiguredTarget: [FileToIndex]] = [:]
     for file in outOfDateFiles {
-      guard let target = await buildSystemManager.canonicalConfiguredTarget(for: file) else {
-        logger.error("Not indexing \(file.forLogging) because the target could not be determined")
+      guard let target = await buildSystemManager.canonicalConfiguredTarget(for: file.uri) else {
+        logger.error("Not indexing \(file.uri.forLogging) because the target could not be determined")
```

Comment:
It probably(?) doesn't matter, but IMO passing the main file here makes more sense since that's the thing we're indexing

Reply:
For `canonicalConfiguredTarget(for:)`: Yes, you’re right, we should use `file.mainFile ?? file.uri`

For `logger.error`: Added both `file.uri` and `file.mainFile` to the logging message.